src/main/java/fr/sncf/osrd/train/TrainPhysicsIntegrator.java

```diff
@@ -172,163 +162,55 @@ private static double computeTimeDelta(double currentSpeed, double acceleration,
-    }
+    public IntegrationStep computeUpdateFromAction(Action action, double maxDistance, double directionSign) {
 
-    public PositionUpdate computeUpdate(Action action, double maxDistance) {
-        return computeUpdate(action, maxDistance, 1);
-    }
-
-    public PositionUpdate computeUpdate(double actionTractionForce, double actionBrakingForce, double maxDistance) {
-        return computeUpdate(actionTractionForce, actionBrakingForce, maxDistance, 1);
+        return computeUpdateFromAcceleration(
+                computeAcceleration(action),
+                action.tractionForce(),
+                maxDistance,
+                directionSign
+        );
     }
 
     /**
-     * Compute the train's acceleration given an action force
-     * @param actionTractionForce the force directly applied by the driver
-     * @param actionBrakingForce the braking force directly applied by the driver
+     * Compute the train's next integration step given an acceleration
+     * @param acceleration the acceleration of the train during that step
+     * @param tractionForce the traction force of the train during that step
      * @param maxDistance the maximum distance the train can go
-     * @param directionSign +1 if the train is going forward, -1 if the train is going backwards
+     * @param directionSign +1 if forward circulation, -1 if backwards
      * @return the new speed of the train
      */
-    public PositionUpdate computeUpdate(double actionTractionForce, double actionBrakingForce, double maxDistance,
-                                        double directionSign) {
-        assert actionBrakingForce >= 0.;
-
-        // the sum of forces that always go the direction opposite to the train's movement
-        double oppositeForce = rollingResistance + actionBrakingForce;
-
-        // as the oppositeForces is a reaction force, it needs to be adjusted to be opposed to the other forces
-        double effectiveOppositeForces;
-        if (currentSpeed == 0.0) {
-            var totalOtherForce = computeTotalForce(0.0, actionTractionForce);
-            // if the train is stopped and the rolling resistance is greater in amplitude than the other forces,
-            // the train won't move
-            if (abs(totalOtherForce) < oppositeForce)
-                return new PositionUpdate(timeStep, 0.0, 0.0, 0.0, 0.0);
-            // if the sum of other forces isn't sufficient to keep the train still, the rolling resistance
-            // will be opposed to the movement direction (which is the direction of the other forces)
-            effectiveOppositeForces = Math.copySign(oppositeForce, -totalOtherForce);
-        } else {
-            // if the train is moving, the rolling resistance if opposed to the movement
-            effectiveOppositeForces = Math.copySign(oppositeForce, -currentSpeed);
-        }
+    public IntegrationStep computeUpdateFromAcceleration(double acceleration, double tractionForce,
```

Comment:
change the name of the method with computeIntegrationStepFromAcceleration

Reply:
Done !